cmdline-pull/discovery_doc.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
```

Comment:
I'm not a fan of having two files named the same thing that do mostly the same thing. Can you consolidate this into one file and make the caching mechanism a parameter ('datastore', 'file')?